### PR TITLE
[VCARB-78] add support for schema validation for server/import using JSON schema

### PR DESCRIPTION
**Pull request by jhaynie**
Initial implementation of using a JSON schema to validate both import and event processing.

Set using `--schema-validator`

Example:

```
go run . import --url "s3://localhost:4566/test" --schema-validator ./datadir/schema --api-key $TOKEN  --verbose --data-dir ./datadir
```

Create a folder schema with the following:

```json
{
  "labor": {
    "schema": "labor_message.json",
    "path": "{{.table}}/received/{{.table}}_{{.mvccTimestamp}}_{{.id}}.json"
  },
  "message": {
    "schema": "base/message.json",
    "path": "{{.table}}/received/{{.table}}_{{.mvccTimestamp}}_{{.id}}.json"
  }
}
```

And copy the test files from `./internal/util/testdata` into this folder

### CODE REVIEW

File: internal/drivers/s3/s3.go [hunk 1]
```diff
@@ -148,14 +165,26 @@ func (p *s3Driver) connect(ctx context.Context, urlString string) error {
 		o.UsePathStyle = true
 	})
 
+	if _, err := p.s3.ListObjects(ctx, &awss3.ListObjectsInput{Bucket: aws.String(p.bucket), MaxKeys: aws.Int32(1)}); err != nil {
+		var bnf *types.NoSuchBucket
+		if errors.As(err, &bnf) {
+			if _, err := p.s3.CreateBucket(ctx, &awss3.CreateBucketInput{Bucket: aws.String(p.bucket)}); err != nil {
```
Comment (byrider-jparker4) [inline internal/drivers/s3/s3.go:159]: Just an outsiders perspective, but I would recommend **against** giving it the ability to create buckets unless there’s an “opt-in” flag or something to allow for more purposeful configuration. That could very easily lead to misconfigurations where sensitive data could end up in buckets that don’t have the appropriate security provisions on it. And if this tool is responsible for doing so, could that have liability implications?

Comment (jhaynie) [reply internal/drivers/s3/s3.go:159]: yes, i will consider that. however, by default most systems require you to explicitly given these elevated permissions.

Comment (jhaynie) [reply internal/drivers/s3/s3.go]: OK, @byrider-jparker4 good feedback. just changed it to only create when testing against a local s3 bucket (something like local stack).

File: internal/schema.go [hunk 2]
```diff
@@ -43,3 +43,9 @@ type SchemaRegistry interface {
 	// Save the latest schema to a file.
 	Save(filename string) error
 }
+
+// SchemaValidator is the interface for a schema validator.
+type SchemaValidator interface {
+	// Validate the event against the schema.
```
Comment (robindiddams) [inline internal/schema.go:49]:
```suggestion
	// Validate the event against the schema. Returns true if a schema is found for the table, true if the event is valid, the path transformed and an error if one occurs.
```

**Review by robindiddams — APPROVED** [review]

**Pull request merged**